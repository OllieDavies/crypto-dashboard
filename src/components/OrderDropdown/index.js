import React from "react";
import cx from "classnames";
import styles from "./styles.module.css";

const OrderDropdown = ({ options, onChange, containerClassName }) => {
  return (
    <select
      data-testid="orderdropdownSelect"
      className={cx(containerClassName, styles.select)}
      onChange={(event) => onChange(event.target.value)}
    >
      {Object.entries(options).map((value) => (
        <option key={value[0]} value={value[0]}>
          {value[1]}
        </option>
      ))}
    </select>
  );
};

export default OrderDropdown;
