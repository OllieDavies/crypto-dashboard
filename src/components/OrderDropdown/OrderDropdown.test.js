import React from "react";
import { render, fireEvent } from "@testing-library/react";

import OrderDropdown from ".";

describe("<OrderDropdown />", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <OrderDropdown options={{ asc: "High to Low", desc: "Low to High" }} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onChange when new item is selected", async () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(
      <OrderDropdown
        options={{ asc: "High to Low", desc: "Low to High" }}
        onChange={mockFn}
      />
    );

    fireEvent.change(getByTestId("orderdropdownSelect"), {
      target: { value: "desc" },
    });
    expect(mockFn).toHaveBeenCalledTimes(1);
    expect(mockFn).toHaveBeenCalledWith("desc");
  });
});
