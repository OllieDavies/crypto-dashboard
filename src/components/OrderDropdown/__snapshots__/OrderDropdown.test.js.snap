// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDropdown /> matches snapshot 1`] = `
<select
  class="select"
  data-testid="orderdropdownSelect"
>
  <option
    value="asc"
  >
    High to Low
  </option>
  <option
    value="desc"
  >
    Low to High
  </option>
</select>
`;
